vertexai/snippets/src/main/java/vertexai/gemini/AutomaticFunctionCalling.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vertexai.gemini;
+
+// [START generativeaionvertexai_gemini_automatic_function_calling]
+// [START aiplatform_gemini_automatic_function_calling]ß
+import com.google.cloud.vertexai.VertexAI;
+import com.google.cloud.vertexai.api.FunctionDeclaration;
+import com.google.cloud.vertexai.api.GenerateContentResponse;
+import com.google.cloud.vertexai.api.Tool;
+import com.google.cloud.vertexai.generativeai.AutomaticFunctionCallingResponder;
+import com.google.cloud.vertexai.generativeai.ChatSession;
+import com.google.cloud.vertexai.generativeai.FunctionDeclarationMaker;
+import com.google.cloud.vertexai.generativeai.GenerativeModel;
+import com.google.cloud.vertexai.generativeai.ResponseHandler;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+public class AutomaticFunctionCalling {
+  public static void main(String[] args) throws IOException, NoSuchMethodException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-google-cloud-project-id";
+    String location = "us-central1";
+    String modelName = "gemini-1.5-flash-001";
+
+    String promptText = "What's the weather like in Paris?";
+
+    whatsTheWeatherLike(projectId, location, modelName, promptText);
+  }
+
+  /** Callable function getCurrentWeather. */
```

Comment:
Can we add a comment clarifying this is just a stub and external functions with http calls can be substituted here?